pyproject.toml

```
@@ -30,6 +30,10 @@ classifiers = [
   "Topic :: Scientific/Engineering :: Visualization",
 ]
 
+dependencies = [
+  "pybind11_stubgen"
+]
```

this will download and install `pybind11_stubgen` when doing `pip install f3d` which isn't necessary for normal use

there is an extra/optional dependency mechanism that could be used so that `pybind11_stubgen` is only installed on `pip install f3d[dev]`
```suggestion
[project.optional-dependencies]
dev = [
  "pybind11_stubgen"
]
```
I'm not sure what the exact workflow is tho (build wheel vs build bindings from cmake vs install from source), maybe it's not necessary at all because it has to be installed externally first anyway?

---

Yeah I'm not sure. It's a build dependency, so it's needed if you type `pip install .` from f3d repository but not needed if installed using wheels.  
I'll try to find the correct way to do it, but maybe it's `optional-dependencies` as you said.

---

there's also
```
[build-system]
requires = ["scikit-build-core", "setuptools-scm>=8.0"]
```
maybe it should go in there?

---

yeah let's try that.

---

indeed it works, thanks :)